addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/handler/MilightBridgeHandler.java

```diff
@@ -0,0 +1,251 @@
+     * and is send as response for a discovery message.
+     */
+    @Override
+    public void initialize() {
+        thingDiscoveryService = new ThingDiscoveryService(thing.getUID(), this);
+        thingDiscoveryService.start(bundleContext);
+
+        createCommunicationObject();
+        setupRefrehTimer();
+    }
+
+    private void createCommunicationObject() {
+        Object host_config_obj = thing.getConfiguration().get(MilightBindingConstants.CONFIG_HOST_NAME);
+        String host_config = ((host_config_obj instanceof String) ? (String) host_config_obj
+                : (host_config_obj instanceof InetAddress) ? ((InetAddress) host_config_obj).getHostAddress() : null);
+
+        InetAddress addr = null;
+
+        try {
+            addr = InetAddress.getByName(host_config);
+        } catch (UnknownHostException ignored) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No address known!");
+            return;
+        }
+
+        bridgeid = (String) thing.getConfiguration().get(MilightBindingConstants.CONFIG_ID);
+
+        // Version 1/2 do not support response messages / detection. We therefore directly call bridgeDetected(addr).
+        if (bridgeid == null || bridgeid.length() != 12) {
+            bridgeid = null; // for the case length() != 12
+            logger.warn("BridgeID not known. Version 2 fallback behaviour activated, no periodical refresh available!");
+            bridgeDetected(addr, "");
+            return;
+        }
+
+        // The MilightDiscover class is used here for periodically ping the device and update the state.
+        if (discover != null) {
+            discover.stopReceiving();
+        }
+        try {
+            discover = new MilightDiscover(addr, this, 1000, 3);
+            discover.start();
+        } catch (SocketException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            return;
+        }
+
+        discover.sendDiscover(scheduler);
+    }
+
+    /**
+     * Sets up the periodically refresh via the scheduler. If the user set CONFIG_REFRESH to 0, no refresh will be
+     * done.
+     */
+    private void setupRefrehTimer() {
+        // Version 1/2 do not support response messages / detection.
+        if (bridgeid == null) {
+            return;
+        }
+
+        if (discoverTimer != null) {
+            discoverTimer.cancel(true);
+        }
+
+        BigDecimal interval_config = (BigDecimal) thing.getConfiguration().get(MilightBindingConstants.CONFIG_REFRESH);
+        if (interval_config == null || interval_config.intValue() == 0) {
+            refrehInterval = 0;
+            return;
+        }
+
+        refrehInterval = interval_config.intValue();
+
+        // This timer will do the state update periodically.
+        discoverTimer = scheduler.scheduleAtFixedRate(new TimerTask() {
```

Comment:
Of course, it is still a TimerTask because it was used with a Timer before but i'll change it to a Runnable now